### PR TITLE
LPS-188504 Hide fields that don't work with proxy objects

### PR DESCRIPTION
**Pull request by kildarep**
Hello Reviewer!

Related Issue: https://liferay.atlassian.net/browse/LPS-188504

### CODE REVIEW

**Comment by kildarep** [comment]
ci:test:sf

**Comment by kildarep** [comment]
ci:test:object

**Comment by kildarep** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java [hunk 1]
```diff
@@ -152,16 +153,28 @@ public String[] getForbiddenLastCharacters() {
...
+					(!objectDefinition.getStorageType(
+					).equals(
+						ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE
+					) ||
+					 (!StringUtil.equals(
+						 objectFieldBusinessType.getName(),
+						 ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION) &&
+					  !StringUtil.equals(
+						  objectFieldBusinessType.getName(),
+						  ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT)))));
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java:177]: @kildarep 

We have a method called `isVisible()` from the interface `ObjectFieldBusinessType`. The ideal solution is using it to say if that type will be visible or not. 

You can check the implementation on `EncryptedObjectFieldBusinessType` as an example. It'll be quite similar for the aggregation and attachment types.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 2]
```diff
@@ -1362,6 +1364,26 @@ private void _validateName(
 		}
 	}
 
+	private void _validateObjectDefinitionStorageType(
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:1367]: Better call this method `_validateBusinessType` and then you can put the call to `_validateBusinessTypeEncrypted` inside it.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 3]
```diff
@@ -1362,6 +1364,26 @@ private void _validateName(
...
+		if (objectDefinition.getStorageType(
+			).equals(
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE
+			) &&
+			(businessType.equals(
+				ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION) ||
+			 businessType.equals(
+				 ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT))) {
+
+			throw new ObjectFieldBusinessTypeException(
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:1380]: "Aggregation and Attachment business types are not supported for the Salesforce storage type"

**Review by gabrielwas — CHANGES_REQUESTED** [review]
@kildarep Check my comments, pls. Thx!

**Comment by kildarep** [comment]
Hi @gabrielwas, thanks for the review. I made the changes requested, I believe that's what you asked for, anything just let me know

**Comment by kildarep** [comment]
ci:test:sf

**Comment by kildarep** [comment]
ci:test:relevant

**Comment by kildarep** [comment]
ci:stop

**Comment by kildarep** [comment]
ci:test:sf

**Comment by kildarep** [comment]
ci:test:relevant

**Review by gabrielwas — CHANGES_REQUESTED** [review]
Hi @kildarep!

Can you remove the line 788 (_validateBusinessTypeEncrypted(objectDefinitionId, businessType)) from the file [ObjectFieldLocalServiceImpl.java](https://github.com/liferay-objects/liferay-portal/pull/2350/files#diff-a50d68735e9840b6a835fed264e1b77700dcf952afe1eb61018ab8462a691a16)?

This check is being done inside _validateBusinessType, do you agree? So this call isn't necessary anymore. 

thx

**Comment by kildarep** [comment]
Hi @gabrielwas , 
Removed, the call is really not necessary. 
Thanks for the review

**Review by gabrielwas — APPROVED** [review]
LGTM! :) Thanks @kildarep!

**Comment by gabrielwas** [comment]
ci:test:sf

**Comment by gabrielwas** [comment]
ci:test:relevant

**Pull request closed without merging**